docs/source/llms/llm-evaluate/index.rst

```diff
@@ -0,0 +1,521 @@
+                "provides libraries for various tasks such as data ingestion, processing, and analysis "
+                "through its components like Spark SQL for structured data, Spark Streaming for "
+                "real-time data processing, and MLlib for machine learning tasks",
+            ],
+        }
+    )
+
+    with mlflow.start_run() as run:
+        system_prompt = "Answer the following question in two sentences"
+        # Wrap "gpt-4" as an MLflow model.
+        logged_model_info = mlflow.openai.log_model(
+            model="gpt-4",
+            task=openai.ChatCompletion,
+            artifact_path="model",
+            messages=[
+                {"role": "system", "content": system_prompt},
+                {"role": "user", "content": "{question}"},
+            ],
+        )
+
+        # Use predefined question-answering metrics to evaluate our model.
+        results = mlflow.evaluate(
+            logged_model_info.model_uri,
+            eval_data,
+            targets="ground_truth",
+            model_type="question-answering",
+        )
+        print(f"See aggregated evaluation results below: \n{results.metrics}")
+
+        # Evaluation result for each data record is available in `results.tables`.
+        eval_table = results.tables["eval_results_table"]
+        print(f"See evaluation table below: \n{eval_table}")
+
+
+LLM Evaluation Metrics
+=======================
+
+There are two types of LLM evaluation metrics in MLflow:
+
+1. Metrics relying on SaaS model (e.g., OpenAI) with prompt engineering, e.g., :py:func:`mlflow.metrics.relevance`. These  
+   metrics are created via :py:func:`mlflow.metrics.make_genai_metric` method. For each data record, these metrics under the hood sends 
+   one prompt consisting of the following information to the SaaS model, and extract the score from model response:
+
+   * Metrics definition.
+   * Metrics grading criteria.
+   * Reference examples.
+   * Input data/context.
+   * Model output.
+   * [optional] Ground truth.
+
+   More details of how these fields are set can be found in the section "Create your Custom LLM-evaluation Metrics".
+
+2. Function-based per-row metrics. These metrics calculate a score for each data record (row in terms of Pandas/Spark dataframe),
+   based on certain functions, like Rouge (:py:func:`mlflow.metrics.rougeL`) or Flesch Kincaid (:py:func:`mlflow.metrics.flesch_kincaid_grade_level`). 
+   These metrics are similar to traditional metrics.
+
+
+Select Metrics to Evaluate
+--------------------------
+
+MLflow LLM evaluation includes default collections of metrics for pre-selected tasks, e.g, "question-answering". Depending on the 
+type of LLM use case that you are evaluating, these pre-defined collections can greatly simplify the process of running evaluations. 
+
+The supported metrics for given model types are shown below:
+
+* **question-answering**:
```

Comment:
token count is included for all 3 of these types. can we add that

Reply:
hmm that's not evaluation metrics right? I feel it's just some extra information computed along with evaluating.

Reply:
while it may be a bit different semantically, its computed as an evaluation metric so i think we should include it here.